### PR TITLE
Revisions based on comments from Colon Pull request #43

### PR DESCRIPTION
**Pull request by mlin865**
Pull request #43 will be cancelled in view of revisions made on new branch.

### CODE REVIEW

File: scaffoldmaker/meshtypes/meshtype_3d_haustra1.py [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+        haustraInnerRadiusFactor = options['Haustra inner radius factor']
+        haustrumSegmentEndDerivativeFactor = options['Haustrum segment end-derivative factor']
+        haustrumSegmentMidDerivativeFactor = options['Haustrum segment mid-derivative factor']
+        wallThickness = options['Wall thickness']
+        haustraSegmentLength = options['Haustra segment length']
+        useCrossDerivatives = options['Use cross derivatives']
+        useCubicHermiteThroughWall = not(options['Use linear through wall'])
+        elementsCountAlong = elementsCountAlongHaustrum
+
+        nodeIdentifier = 1
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_haustra1.py]: The remainder of this function should call getColonHaustraSegmentInnerPoints and then generate with the tube mesh - copy the code from 'Colon 1'.
You just have to make a centre line:
x = [  [  0.0, 0.0, 0.0 ], [ haustrumLength, 0.0, 0.0 ] ]
d1 = [  [ haustrumLength, 0.0, 0.0 ], [ haustrumLength, 0.0, 0.0 ]  ]

File: scaffoldmaker/meshtypes/meshtype_3d_colon1.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            'Default',
+            'Human 1']
+
+    @staticmethod
+    def getDefaultOptions(parameterSetName='Default'):
+        options = MeshType_3d_haustra1.getDefaultOptions(parameterSetName)
+        options['Number of elements around'] = 15
+        options['Number of elements along haustrum'] = 4
+        options['Inner radius'] = 1.0
+        options['Haustrum segment mid-derivative factor'] = 2.0
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colon1.py]: Rename consistently with 'Haustra 1' scaffold... see later.
'Haustrum length mid derivative factor'

File: scaffoldmaker/meshtypes/meshtype_3d_colon1.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            }
+        options.update(optionsColon)
+        if 'Human 1' in parameterSetName:
+            options['Tube type'] = 3
+        return options
+
+    @staticmethod
+    def getOrderedOptionNames():
+        optionNames = MeshType_3d_haustra1.getOrderedOptionNames()
+        optionNames.remove('Haustra segment length')
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colon1.py]: 'Haustrum length'

File: scaffoldmaker/meshtypes/meshtype_3d_colon1.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        :param options: Dict containing options. See getDefaultOptions().
+        :return: annotationGroups
+        """
+        elementsCountAround = options['Number of elements around']
+        elementsCountAlongHaustrum = options['Number of elements along haustrum']
+        elementsCountThroughWall = options['Number of elements through wall']
+        haustraSegmentCount = options['Number of haustra segments']
+        radius = options['Inner radius']
+        cornerInnerRadiusFactor = options['Corner inner radius factor']
+        haustraInnerRadiusFactor = options['Haustra inner radius factor']
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colon1.py]: Carefully update option names and variable names - see comments in 'Haustra 1' scaffold.

File: scaffoldmaker/meshtypes/meshtype_3d_colon1.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+        # find arclength of colon
+        length = 0.0
+        elementsCountIn = len(cx) - 1
+        sd1 = smoothCubicHermiteDerivativesLine(cx, cd1, fixAllDirections = True,
+            magnitudeScalingMode = DerivativeScalingMode.HARMONIC_MEAN)
+        for e in range(elementsCountIn):
+            arcLength = getCubicHermiteArcLength(cx[e], sd1[e], cx[e + 1], sd1[e + 1])
+            length += arcLength
+        haustraSegmentLength = length / haustraSegmentCount
```
Comment (rchristie) [inline scaffoldmaker/meshtypes/meshtype_3d_colon1.py]: haustrumLength

**Comment by mlin865** [comment]
I've made the requested changes and tested on both haustra and colon scaffolds to ensure they are running properly.

**Pull request merged**